test/appium/tests/atomic/chats/test_public.py

```diff
@@ -328,6 +329,304 @@ def test_public_chat_open_using_deep_link(self):
             self.driver.fail("Public chat '%s' is not opened" % chat_name)
 
 
+@pytest.mark.xdist_group(name="messaging_1")
+class TestMessaging(MultipleSharedDeviceTestCase):
+
+    @classmethod
+    def setup_class(cls):
+        cls.drivers, cls.loop = create_shared_drivers(2)
+        cls.device_1, cls.device_2 = SignInView(cls.drivers[0]), SignInView(cls.drivers[1])
+        cls.home_1, cls.home_2 = cls.device_1.create_user(enable_notifications=True), cls.device_2.create_user()
+        cls.public_key_1, cls.default_username_1 = cls.home_1.get_public_key_and_username(return_username=True)
+        cls.public_key_2, cls.default_username_2 = cls.home_2.get_public_key_and_username(return_username=True)
+        profile_2 = cls.home_2.get_profile_view()
+        profile_2.switch_network()
+        [home.home_button.click() for home in (cls.home_1, cls.home_2)]
+        cls.chat_2 = cls.home_2.add_contact(cls.public_key_1)
+        cls.chat_2.chat_message_input.send_keys('test message')
+        cls.chat_2.send_message_button.click()
+        cls.home_2.just_fyi('Install free sticker pack and use it in 1-1 chat')
+        cls.chat_2.install_sticker_pack_by_name()
+        cls.chat_1 = cls.home_1.add_contact(cls.public_key_2)
+        [home.home_button.click() for home in (cls.home_1, cls.home_2)]
+        cls.group_chat_name = "GroupChat"
+        cls.group_chat_1 = cls.home_1.create_group_chat(user_names_to_add=[cls.default_username_2], group_chat_name=cls.group_chat_name)
+        cls.home_2.get_chat(cls.group_chat_name).click()
+        cls.group_chat_2 = cls.home_2.get_chat_view()
+        cls.group_chat_2.join_chat_button.click()
+        [home.home_button.click() for home in (cls.home_1, cls.home_2)]
+        cls.public_chat_name = cls.home_1.get_random_chat_name()
+        cls.public_chat_1, cls.public_chat_2 = cls.home_1.join_public_chat(cls.public_chat_name), cls.home_2.join_public_chat(
+            cls.public_chat_name)
+        [home.home_button.click() for home in (cls.home_1, cls.home_2)]
+        cls.home_1.get_chat(cls.default_username_2).click()
+        cls.home_2.get_chat(cls.default_username_1).click()
+        cls.message_1, cls.message_2, cls.message_3, cls.message_4 = "Message1", "Message2", "Message3", "Message4"
```

Comment:
IMO better to divide it into blocks (creating 1-1, public, group chat)

Reply:
fixed

Reply:
to see this in logs, better to add cls.home_2.just_fyi; in that case you'll see clearly if smth go wrong

Reply:
yeah, that is exactly what I did